language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/changetype/FixReturnTypeCodeAction.java

```diff
@@ -97,28 +99,43 @@ public List<CodeAction> getDiagBasedCodeActions(Diagnostic diagnostic,
             return Collections.emptyList();
         }
 
+        List<TextEdit> importEdits = new ArrayList<>();
+        List<String> types = new ArrayList<>();
+        boolean returnsClause = funcDef.get().functionSignature().returnTypeDesc().isPresent();
+
+        if (checkExprDiagnostic) {
+            // Add error return type for check expression
+            String error = "error";
+            String optionalError = "error?";
+            if (returnsClause) {
+                types.add(funcDef.get().functionSignature().returnTypeDesc().get().type().toString().trim().concat("|")
```

Comment:
What if the existing return type is an optional type like `int?`? The resulting return type will be `int?|error?` right?

Reply:
No, since `returnTypeDescPresent` will return true, only `error` will get added. Test case for this : [fixReturnTypeWithCheckExpr4.json](https://github.com/ballerina-platform/ballerina-lang/pull/36040/files#diff-e5e1aa48d475083ef42d5cc74d4eecaf9cea539cae03fc2bbca37942224d2321)

Reply:
Still the return type will be `int?|error` right? I was thinking whether we can improve this to rewrite as `int|error|()` or `int|error?` (all are identical)

Reply:
Yes, the return type will be `int?|error`. Since they're completely identical, it's unclear to me what the improvement/ difference will be, by rewriting `int?|error` as `int|error?` or `int|error|()`. Won't it simply be complicating the logic?

Reply:
Let's keep this as it is.